diff --git a/subprojects/gst-plugins-bad/ext/srtp/meson.build b/subprojects/gst-plugins-bad/ext/srtp/meson.build
index 7f947191ae..b4a8e9c2a8 100644
--- a/subprojects/gst-plugins-bad/ext/srtp/meson.build
+++ b/subprojects/gst-plugins-bad/ext/srtp/meson.build
@@ -6,12 +6,14 @@   'gstsrtpdec.c',
   'gstsrtpenc.c',
 ]
 
+gst_plugins_install_dir = join_paths(get_option('libdir'), 'gstreamer-1.0')
+
 srtp_cargs = []
 if get_option('srtp').disabled()
   subdir_done()
 endif
 
-srtp_dep = dependency('libsrtp2', version : '>= 2.1.0', required : false)
+srtp_dep = dependency('libSRTP', modules: ['libSRTP::srtp2'], version : '>= 2.1.0', required : false)
 if srtp_dep.found()
   srtp_cargs += ['-DHAVE_SRTP2']
 else
@@ -37,6 +39,6 @@     link_args : noseh_link_args,
     include_directories : [configinc],
     dependencies : [gstrtp_dep, gstvideo_dep, srtp_dep],
     install : true,
-    install_dir : plugins_install_dir,
+    install_dir : gst_plugins_install_dir,
   )
   plugins += [gstsrtp]
